### PR TITLE
Show recommended badge on details page

### LINKED ISSUE #7794: Show recommended badge on details page
UX is as follows:

Screenshots

Depends on changes in https://github.com/orgs/mozilla/projects/68

See https://mozilla.invisionapp.com/share/BXQTIIC86ZN for the latest UX

TBD: There should be a tooltip that pops up when hovering over the badge, explaining what it means.

TBD: The badge should link to a SUMO article explaining the recommendation process.

Note also that the Recommended badge should not appear for Android.

### PR DESCRIPTION
**Pull request by bobsilverberg**
~~Depends on #7958~~

Fixes #7794 

Large screen:

Screenshot 2019-05-02 15 47 05

Small screen:

Screenshot 2019-05-02 15 48 00

I know there are some issues with the small screen, but I think I'll be able to address them once I have the correct icon.

@jvillalobos & @brassy- Do we want the Recommended badge to appear lower down on the small version of the screen, as it does in my screenshot above? I've simply put it above all the other badges on both the large and small screens. But maybe we want this new Recommended badge to show up in the upper right-hand corner even on smaller screens?

### CODE REVIEW

**Comment by jvillalobos** [comment]
I suppose it should appear in the same location as all badges. This question reminded me that we won't need the Featured Extension badge after this, so I filed https://github.com/mozilla/addons-frontend/issues/7960

**Comment by bobsilverberg** [comment]
@willdurand I saw you asked about displaying these badges for Android in IRC and @jvillalobos indicated that we should not. I added a commit to exclude the Recommended badge on Android.

File: src/amo/components/AddonBadges/index.js [hunk 1]
```diff
@@ -1,80 +1,109 @@
...
+      isQuantumCompatible({ addon }) === false;
+
+    return (
+      <div className="AddonBadges">
+        {_config.get('enableFeatureRecommendedBadges') &&
+        addon.is_recommended &&
+        clientApp !== CLIENT_APP_ANDROID ? (
+          <RecommendedBadge />
+        ) : null}
+        {addon.is_featured ? (
```
Comment (willdurand) [inline src/amo/components/AddonBadges/index.js]: Shouldn't we hide this badge when `enableFeatureRecommendedBadges` is `true` and we are not on Android?

Comment (willdurand) [reply src/amo/components/AddonBadges/index.js]: See also: https://github.com/mozilla/addons-frontend/issues/7960

Comment (bobsilverberg) [reply src/amo/components/AddonBadges/index.js]: Exactly. I was going to defer that to #7960, but if you think I should just do it now I can.

Comment (willdurand) [reply src/amo/components/AddonBadges/index.js]: I think we could do that here, no?

Comment (bobsilverberg) [reply src/amo/components/AddonBadges/index.js]: Sure, why not? :-)

**Comment by bobsilverberg** [comment]
@willdurand I've hidden the Featured badge for extensions if the `enableFeatureRecommendedBadges` is on.

File: tests/unit/amo/components/TestAddonBadges.js [hunk 2]
```diff
@@ -42,18 +50,113 @@ describe(__filename, () => {
...
+      _config: getFakeConfig({
+        enableFeatureRecommendedBadges: true,
+      }),
+      addon,
+    });
+
+    expect(root.find(Badge)).toHaveProp('type', 'featured');
+  });
+
+  it('does not display a badge for an extension when the addon is featured and the recommended feature is on', () => {
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonBadges.js]: What if the extension is both featured and recommended? Do we need to write a test case for that?

File: tests/unit/amo/components/TestAddonBadges.js [hunk 3]
```diff
@@ -42,18 +50,113 @@ describe(__filename, () => {
     expect(root.find(Badge)).toHaveLength(0);
   });
 
-  it('displays a badge when the addon is featured', () => {
+  it('displays a badge when the addon is recommended', () => {
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonBadges.js]: a _recommended_ badge

File: tests/unit/amo/components/TestAddonBadges.js [hunk 4]
```diff
@@ -42,18 +50,113 @@ describe(__filename, () => {
...
+        enableFeatureRecommendedBadges: false,
+      }),
+      addon,
+    });
 
     expect(root.find(Badge)).toHaveProp('type', 'featured');
     expect(root.find(Badge)).toHaveProp('label', 'Featured Extension');
   });
 
+  it('displays a badge for a theme when the addon is featured and the recommended feature is on', () => {
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonBadges.js]: a _featured_ badge

File: tests/unit/amo/components/TestAddonBadges.js (hunk 2 above)
Comment (bobsilverberg) [reply tests/unit/amo/components/TestAddonBadges.js]: I don't think we do. The two are unrelated, i.e., the `is_recommended` flag has nothing to do with whether we show the Featured badge, and vice versa. It seems like an unnecessary test to me.

File: tests/unit/amo/components/TestAddonBadges.js [hunk 5]
```diff
@@ -42,18 +50,113 @@ describe(__filename, () => {
...
+
+  it('does not display a recommended badge when the feature is disabled', () => {
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      is_recommended: true,
+      type: ADDON_TYPE_EXTENSION,
+    });
+    const root = shallowRender({
+      _config: getFakeConfig({
+        enableFeatureRecommendedBadges: false,
```
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply tests/unit/amo/components/TestAddonBadges.js]: Good point. I changed the default `clientApp` to `CLIENT_APP_FIREFOX` as this was not the only test affected.

**Comment by bobsilverberg** [comment]
Thanks @willdurand. I have pushed a new commit.

**Pull request merged**